engine/src/main/java/com/datatorrent/stram/plan/logical/LogicalPlanConfiguration.java

```diff
@@ -124,7 +153,727 @@ public static StramElement fromValue(String value) {
+
+        if (result != null) {
+          return result;
+        }
+      }
+
+      return null;
+    }
+
+    protected static Conf addConfs(Conf parentConf, ConfElement childConfElement)
+    {
+      //Figure out what configurations need to be added to hold this attribute
+      List<StramElement> path = ConfElement.getPathFromParentToChildInclusive(childConfElement.getStramElement(),
+                                                                              parentConf.getConfElement().getStramElement());
+
+      for (int pathIndex = 1;
+           pathIndex < path.size();
+           pathIndex++) {
+        LOG.debug("Adding conf");
+        StramElement pathElement = path.get(pathIndex);
+        //Add the configurations we need to hold this attribute
+        parentConf = addConf(pathElement, WILDCARD, parentConf);
+      }
+
+      return parentConf;
+    }
+
+  }
+
+  /**
+   * Utility class that holds methods for handling {@link Context} classes.
+   */
+  @SuppressWarnings("unchecked")
+  protected static class ContextUtils
+  {
+    public static final Map<Class<? extends Context>, Set<String>> CONTEXT_CLASS_TO_ATTRIBUTES;
+    public static final Set<Class<? extends Context>> CONTEXT_CLASSES;
+    public static final Map<Class<? extends Context>, Map<String, Attribute<?>>> CONTEXT_TO_ATTRIBUTE_NAME_TO_ATTRIBUTE;
+
+    static {
+      CONTEXT_CLASSES = Sets.newHashSet();
+
+      for (Class<?> clazz: Context.class.getDeclaredClasses()) {
+        if (!Context.class.isAssignableFrom(clazz)) {
+          continue;
+        }
+
+        CONTEXT_CLASSES.add((Class<? extends Context>)clazz);
+      }
+
+      CONTEXT_CLASS_TO_ATTRIBUTES = Maps.newHashMap();
+
+      for (Class<? extends Context> contextClass: CONTEXT_CLASSES) {
+        Set<String> contextAttributes = Sets.newHashSet();
+
+        Field[] fields = contextClass.getDeclaredFields();
+
+        for (Field field: fields) {
+          if (!Attribute.class.isAssignableFrom(field.getType())) {
+            continue;
+          }
+
+          contextAttributes.add(field.getName());
+        }
+
+        CONTEXT_CLASS_TO_ATTRIBUTES.put(contextClass, contextAttributes);
+      }
+
+      CONTEXT_TO_ATTRIBUTE_NAME_TO_ATTRIBUTE = Maps.newHashMap();
+
+      for (Class<? extends Context> contextClass: CONTEXT_CLASSES) {
+        Map<String, Attribute<?>> simpleAttributeNameToAttribute = Maps.newHashMap();
+        CONTEXT_TO_ATTRIBUTE_NAME_TO_ATTRIBUTE.put(contextClass, simpleAttributeNameToAttribute);
+
+        Set<Attribute<Object>> attributes = AttributeInitializer.getAttributesNoSave(contextClass);
+
+        LOG.debug("context class {} and attributes {}", contextClass, attributes);
+
+        for (Attribute<Object> attribute: attributes) {
+          simpleAttributeNameToAttribute.put(AttributeParseUtils.getSimpleName(attribute), attribute);
+        }
+      }
+    }
+
+    private ContextUtils()
```

Comment:
This class only contains static methods and should not be instantiated. A private constructor ensures that.